crypto/fipsmodule/ec/ec_test.cc

```diff
@@ -182,6 +182,259 @@ TEST(ECTest, Encoding) {
       y_hex.get());
 }
 
+// P-{224,256,384,521} test vectors, taken from CAVP
+// (CAVP 20.1 - KASValidityTest_ECCStaticUnified_KDFConcat_NOKC)
+// https://csrc.nist.gov/projects/cryptographic-algorithm-validation-program/key-management
+
+static const uint8_t kP224PublicKey_uncompressed[] = {
+  /* uncompressed */
+  0x04,
+  /* x-coordinate */
+  0xd6, 0xf5, 0xf0, 0x6e, 0xf4, 0xc5, 0x56, 0x0a, 0xff, 0x8f, 0x49, 0x90,
+  0xef, 0xdb, 0xa5, 0x9a, 0xf8, 0xa8, 0xd3, 0x77, 0x0d, 0x80, 0x14, 0x6a,
+  0xc5, 0x82, 0x78, 0x85,
+  /* y-coordinate */
+  0xe0, 0x43, 0xae, 0x7b, 0xae, 0xa3, 0x77, 0x28, 0x60, 0x39, 0xc0, 0x7c,
+  0x04, 0x1b, 0x7a, 0x3b, 0x5d, 0x76, 0x96, 0xda, 0xdd, 0xa7, 0x05, 0x1a,
+  0xd6, 0x45, 0xa3, 0xea
+};
+
+static const uint8_t kP224PublicKey_compressed[] = {
+  0x02,
```

Comment:
This encodes the parity of the y-value of the represented point.

I'm not entirely sure how the dec/enc functions work. But do we need more cases that cover both possible values for all curves?

Reply:
> But do we need more cases that cover both possible values for all curves?

Another possible value is `0x03`? This is verified by another curve (e.g. `kP256PublicKey_compressed`).

The problem is searching these tests inputs may take some time. Let me know if you know some test source.

Reply:
I mean, what if the decode path is different for each curve:

For P224 the call path is:
P224 -> decode -> **if** first byte is `0x02` **do** f_224_dec() **else if** first byte is `0x03` **do** `g_224_dec()`.

But for P256 it is completely different:
P256 -> decode -> **if** first byte is `0x02` **do** f_256_dec() **else if** first byte is `0x03` **do** `g_256_dec()`.

So, if there is not a general decode function for all curves, there are paths that are not covered.

Reply:
Added `0x02` and `0x03` test cases.